src/integrationtests/java/com/aws/greengrass/integrationtests/KeystoreTest.java

```diff
@@ -37,21 +44,52 @@
 import static com.aws.greengrass.lifecyclemanager.GreengrassService.SERVICES_NAMESPACE_TOPIC;
 import static com.aws.greengrass.testcommons.testutilities.ExceptionLogProtector.ignoreExceptionOfType;
 import static com.aws.greengrass.testcommons.testutilities.TestUtils.asyncAssertOnConsumer;
+import static com.github.grantwest.eventually.EventuallyLambdaMatcher.eventuallyEval;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @BridgeIntegrationTest
 public class KeystoreTest {
     private static final long AWAIT_TIMEOUT_SECONDS = 30L;
 
+    ExecutorService executor = Executors.newCachedThreadPool();
     BridgeIntegrationTestContext testContext;
 
+    @AfterEach
+    void tearDown() {
+        executor.shutdownNow();
+    }
+
+    @TestWithMqtt5Broker
+    @WithKernel("mqtt5_config_ssl.yaml")
+    void GIVEN_mqtt_bridge_WHEN_client_cert_changes_THEN_local_client_restarts(Broker broker) throws Exception {
+        CountDownLatch clientReset = new CountDownLatch(1);
+        executor.submit(() -> {
+            assertThat("client disconnected", () -> testContext.getLocalV5Client().getClient().getIsConnected(), eventuallyEval(is(false)));
+            assertThat("client reconnected", () -> testContext.getLocalV5Client().getClient().getIsConnected(), eventuallyEval(is(true)));
+            clientReset.countDown();
```

Comment:
this is going to be impossible to debug.

Use the test utilities async assert functionality.

Reply:
👍  will fix

Reply:
fixed